File: allocation.py
######################
# Created by Vaibhav #
######################

# Google Kickstart Allocation Problem 2020

t = int(input())

for x in range(1,t+1):
    n,b = map(int,input().split())
    houses = list(map(int,input().split()))
    houses.sort()
    
    y = 0
    for house in houses:
        if b >= house:
            b -= house
            y += 1
        else:
            break
            
    print('Case #%s: %s' % (x,y))

File: allocation.cpp
/////////////////////////
// Created by Vaibhav //
///////////////////////

# include <bits/stdc++.h>
using namespace std;

int main() {
  int t;
  cin >> t;

  // loop over test cases
  for (int l = 1; l <=t; l++) {
    int n, b;
    cin >> n; cin >> b;

    // input the houses
    int h[n];
    for (int i=0; i<n; i++) {
      cin >> h[i];
    }

    // sort the houses in ascending order
    sort(h,h+n);

    // compute the answer
    int y = 0;

    for (int p = 0; p < n; p++) {
      if (b >= h[p]) {
        b -= h[p];
        y += 1;
      }
    }

    cout << "Case #" << l << ": " << y << endl;

  }
}
